scripts/stubtest.py

```diff
@@ -5,126 +5,201 @@
 """
 
 import importlib
-import json
-import subprocess
 import sys
-from typing import Dict, Any
-from collections import defaultdict
+from typing import Dict, Any, List
+from collections import defaultdict, namedtuple
 
 from mypy import build
 from mypy.build import default_data_dir, default_lib_path, find_modules_recursive
 from mypy.errors import CompileError
-from mypy.nodes import MypyFile, TypeInfo, FuncItem
+from mypy import nodes
 from mypy.options import Options
 
+import dumpmodule
 
-skipped = {
+if sys.version_info < (3, 4):
+    from singledispatch import singledispatch
+else:
+    from functools import singledispatch
+
+# TODO: email.contentmanager has a symbol table with a None node.
+#       This seems like it should not be.
+
+skip = {
     '_importlib_modulespec',
     '_subprocess',
     'distutils.command.bdist_msi',
     'distutils.command.bdist_packager',
     'msvcrt',
     'wsgiref.types',
+    'mypy_extensions',
+    'unittest.mock',  # mock.call infinite loops on inspect.getsourcelines
+                      # https://bugs.python.org/issue25532
+                      # TODO: can we filter only call?
+}
+
+messages = {
```

Comment:
I would use constants like `NOT_IN_RUNTIME = ('{error.stub_type} ....`, instead of dictionary; that way IDE auto-completes and interpreter will identify a typo right away.